Source/WebCore/Modules/url-pattern/URLPatternComponent.cpp

```diff
@@ -85,5 +92,70 @@ bool URLPatternComponent::matchSpecialSchemeProtocol(ScriptExecutionContext& con
     return isSchemeMatch != specialSchemeList.end();
 }
 
+JSC::JSValue URLPatternComponent::componentExec(ScriptExecutionContext& context, StringView comparedString) const
+{
+    Ref vm = context.vm();
+    JSLockHolder lock(vm);
+
+    auto regex = RegExpObject::create(vm, context.globalObject()->regExpStructure(), m_regularExpression.get(), true);
+    return regex->exec(context.globalObject(), JSC::jsString(vm, comparedString));
+}
+
+static String valueToWTFString(JSC::JSValue input)
+{
+    // FIXME: Is there a better, compact way to do this?
```

Comment:
@szewai @cdumez 

Is there a better, shorter way to convert a JSValue to a string?

Reply:
`JSValue::toString()`?

Reply:
`JSValue::toWTFString()` even